py/foundation/geometry.py

```diff
@@ -130,15 +139,19 @@ def max_x(points: Iterable['Point']) -> float:
   def max_y(points: Iterable['Point']) -> float:
```

Comment:
This and all the other geometric methods (`distance`, etc.) of `Point` are now wrong -- they all assume that all input points have the same `page_index`.

The same goes for the `BBox` member methods.